### ISSUE TITLE
Theme issue on Search bar control in .NET MAUI

### ISSUE THREAD

**Issue by Ahamed-Ali** [dotnet/maui#25153]
### Description

When changing the theme from dark to light or light to dark, the search bar control does not update according to the theme changes.

For example, after switching from the dark theme to the light theme, the search bar's underline and search icon remain the same color, failing to adjust to the new theme.

A video demonstration is provided below.

https://github.com/user-attachments/assets/0206a0a4-4b89-4534-8e71-5cd740841b3a

### Steps to Reproduce

Please get the below repo to reproduce the issue

### Link to public reproduction project repository

https://github.com/Ahamed-Ali/SearchBarTheme

### Version with bug

Unknown/Other

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11

**Comment by Thoronion** [comment]
I can confirm this as well on Android 14 using MAUI 8.0.92.

**Comment by h0lg** [comment]
@jfversluis I notice the SearchButton 🔎 not being visible on **Windows** (MAUI **9**) in the **Light** theme variant - and it doesn't matter which theme variant I start on or whether I switch it. Its border is also not visible.

<img width="964" height="86" alt="Image" src="https://github.com/user-attachments/assets/d7e1aab6-82db-4627-b3bb-76e0759ce7fa" />

<img width="965" height="81" alt="Image" src="https://github.com/user-attachments/assets/1be8f117-4c68-4950-ae6b-2d57cc0427fe" />

---

Same issue with **invisible borders and icons** for `Entry`, `Picker` and `Stepper` buttons:

<img width="967" height="479" alt="Image" src="https://github.com/user-attachments/assets/1d0c3870-d439-46bb-8e93-1e539e29b221" />

<img width="947" height="465" alt="Image" src="https://github.com/user-attachments/assets/a5d1a1c5-797d-4abf-935e-8780885b33f2" />

**Comment by jfversluis** [comment]
I think the change that fixes this should land in .NET 9 SR12, so when that is released, let us know if that fixes it!

**Comment by AlleSchonWeg** [comment]
Hi @jfversluis and @kubaflo 
the change not fixes the line colour under the input field. Look on my screenshot. Sorry, difficult to see:

<img width="426" height="82" alt="Image" src="https://github.com/user-attachments/assets/6a09c52b-47aa-4dfd-aa0b-a02a482b2f11" />

**Comment by kubaflo** [comment]
@AlleSchonWeg you're right! Thank you for noticing it! Here's the fix https://github.com/dotnet/maui/pull/32187

**Comment by AlleSchonWeg** [comment]
> [@AlleSchonWeg](https://github.com/AlleSchonWeg) you're right! Thank you for noticing it! Here's the fix [#32187](https://github.com/dotnet/maui/pull/32187)

Is it possible to get the fix for .Net 9?

**Comment by kubaflo** [comment]
> > [@AlleSchonWeg](https://github.com/AlleSchonWeg) you're right! Thank you for noticing it! Here's the fix [#32187](https://github.com/dotnet/maui/pull/32187)
> 
> Is it possible to get the fix for .Net 9?

I think Net9 won't have any more fixes. I'm like 90% sure, though.

**Comment by h0lg** [comment]
All issues from https://github.com/dotnet/maui/issues/25153#issuecomment-3282929626 persist in MAUI 10 on Windows.

**Edit:** After updating the `Styles.xaml` to the one in the updated project template, the search icon and Entry border are _barely visible_ on the new `Page` `BackgroundColor` for the `Light` variant, which changed from `White` to `LightBackground`.

<img width="90" height="59" alt="Image" src="https://github.com/user-attachments/assets/548960db-8a85-4c49-8a68-6bb7172e79f3" />


```xaml
<Style TargetType="Page" ApplyToDerivedTypes="True">
  <Setter Property="Padding" Value="0"/>
  <Setter Property="BackgroundColor" Value="{AppThemeBinding Light={StaticResource LightBackground}, Dark={StaticResource DarkBackground}}" />
</Style>
```
